cpp/src/io/parquet/page_enc.cu

```diff
@@ -359,6 +359,11 @@ __global__ void __launch_bounds__(128)
       // override this_max_page_size if the requested size is smaller
       this_max_page_size = min(this_max_page_size, max_page_size_bytes);
 
+      // subtract size of rep and def level vectors
+      auto num_vals = values_in_page + frag_g.num_values;
+      this_max_page_size -= max_RLE_page_size(col_g.num_def_level_bits(), num_vals) +
```

Comment:
Maybe we should be using `long` for `this_max_page_size` so it doesn't underflow in some odd case.

Reply:
Ok, I'll take a look at that.

Reply:
@vuule so use signed rather than `size_t` so it will underflow rather than wrap around?  Underflow would just guarantee the test immediately following would evaluate to `true`.

Reply:
But it should evaluate to `true`, right? If `this_max_page_size` somehow becomes negative, it means that we can't fit the def and rep levels in the target page size, even without the data.
With `size_t`, `this_max_page_size` will be a huge positive number instead and we'll keep building the page until other conditions are met (e.g. `num_rows` limit).

Reply:
Yes, I wasn't clear.  Underflow will evaluate true, which is what we want.  Overflow will result in an even larger page, which we don't want.  I think we're in agreement 😄